rotkehlchen/chain/ethereum/modules/shutter/decoder.py

```diff
@@ -0,0 +1,127 @@
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ShutterDecoder(GovernableDecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+            protocol=CPT_SHUTTER,
+            proposals_url='https://shutterdao.discourse.group/c/shutter-dao/dao-proposals/15',
+        )
+        self.shu = A_SHU.resolve_to_evm_token()
+
+    def _decode_shutter_claim(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != SHU_CLAIMED:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = vesting_pool_address = None
+        for tx_log in context.all_logs:
+            if (
+                tx_log.address == self.shu.evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == SHUTTER_AIDROP_CONTRACT
+            ):
+                amount = token_normalized_value(token_amount=hex_or_bytes_to_int(tx_log.data), token=self.shu)  # noqa: E501
+                vesting_pool_address = hex_or_bytes_to_address(tx_log.topics[2])
+                break
+
+        if amount is None:
+            log.error(f'Could not find the amount of the SHU claimed in {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+        if vesting_pool_address is None:
+            log.error(f'Could not find the vesting pool address of the SHU claimed in {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.shu,
+            balance=Balance(amount=FVal(amount)),
+            location_label=hex_or_bytes_to_address(context.tx_log.topics[2]),
+            notes=f'Claim {amount} SHU vested from shutter airdrop and deposit it to {vesting_pool_address}',  # noqa: E501
+            counterparty=CPT_SHUTTER,
+            address=context.transaction.to_address,
+        ))
+
+    def _decode_delegation_change(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a change in the delegated address"""
+        from_delegate = hex_or_bytes_to_address(context.tx_log.topics[2])
```

Comment:
The delegator here is the one who delegates. But since the delegator is the vesting contract in all these cases and we can't know who owns the vesting contract we are forced to make an assumption. `context.transaction.from_address` being the user address.

And this assumption is probaly always gonna hold since you are matching this with input data. So the caller, is the ones who owns the vesting contract.